src/utils/configOverride.ts

```diff
@@ -25,25 +24,23 @@ export function getConfigNavigateCallback(navigate: NavigateFunction) {
   };
 }
 
-function getConfigSource(configKey: string | null, settings: Settings | undefined) {
-  if (!settings) {
-    return '';
-  }
+function getConfigSource(configKey: string | null) {
+  const { APP_UNSAFE_ALLOW_ANY_CONFIG_SOURCE, APP_ADDITIONAL_ALLOWED_CONFIG_SOURCES, APP_DEFAULT_CONFIG_SOURCE } = import.meta.env;
 
   // Skip all the fancy logic below if there aren't any other options besides the default anyhow
-  if (!settings.UNSAFE_allowAnyConfigSource && (settings.additionalAllowedConfigSources?.length || 0) <= 0) {
-    return settings.defaultConfigSource;
+  if (!APP_UNSAFE_ALLOW_ANY_CONFIG_SOURCE && (APP_ADDITIONAL_ALLOWED_CONFIG_SOURCES?.split(',')?.length || 0) <= 0) {
```

Comment:
Let's save this refactor for another PR if we decide it's valuable. This one already has a lot of churn and this is creating more potential for conflicts.